### ISSUE TITLE
Cannot upgrade from 18/edge to 19/stable - snap refresh fails

### ISSUE THREAD

**Issue by locknut** [nextcloud-snap/nextcloud-snap#1808]
#### Describe the bug

The following error is observed when trying to refresh from 18/edge  to 19/stable:

```
root@i-eb549fa0:/snap# snap list  nextcloud
Name       Version        Rev    Tracking  Publisher   Notes
nextcloud  18-2021-05-15  27832  18/edge   nextcloud✓  -
root@i-eb549fa0:/snap# snap refresh --channel=19/stable nextcloud
error: cannot perform the following tasks:
- Run post-refresh hook of "nextcloud" snap if present (run hook "post-refresh": Nextcloud doesn't support skipping major versions, you must upgrade to Nextcloud 18 first. Try 'sudo snap refresh nextcloud --channel=18')
root@i-eb549fa0:/snap# 
```

#### To Reproduce
1. Install nextcloud 18/edge using snap - version 18-2021-05-15 
2. Refresh snap to 19/stable
3. See error

#### Expected behavior

Nextcloud successfully upgrades to 19/stable.

#### Screenshots

N/A

#### OS/snapd/snap version

OS: Ubuntu 18.05.5 LTS

```
root@i-eb549fa0:~# snap list nextcloud
Name       Version        Rev    Tracking  Publisher   Notes
nextcloud  18-2021-05-15  27832  18/edge   nextcloud✓  -
root@i-eb549fa0:~# 
```

```
root@i-eb549fa0:~# snap version
snap    2.51.3
snapd   2.51.3
series  16
ubuntu  18.04
kernel  4.15.0-153-generic
root@i-eb549fa0:~# 
```

Bug is confirmed in #1505 - see https://github.com/nextcloud/nextcloud-snap/issues/1505#issuecomment-895401526

**Comment by kyrofa** [comment]
I have indeed confirmed this. It mostly stems from a lack of tests, though, so I'm working on introducing a bash test suite as well so we can cover this kind of logic in tests.

**Comment by kyrofa** [comment]
Okay so I believe this issue is a combination of two:

1. That which is fixed in #1833
2. The fact that the last version built from stable18 actually removed a few `occ` commands the snap was relying on, which made some of the scripts loop with errors, which means the snap never got all the upgrade steps from 17 to 18 run, which made the snap continue thinking you were on 17. I know this because of the error message: "Nextcloud doesn't support skipping major versions, you must upgrade to Nextcloud 18 first". In order to print "Nextcloud 18" there, it takes the current major version and adds 1, which means it thinks you're on 17. You can see this if you run `sudo snap logs -f nextcloud`.
